pkg/reconciler/workload/defaultplacement/defaultplacement_controller.go

```diff
@@ -0,0 +1,226 @@
+	return c, nil
+}
+
+// controller reconciles watches apibinding and creates a default placement in the same workspace
+type controller struct {
+	queue workqueue.RateLimitingInterface
+
+	kcpClusterClient kcpclient.ClusterInterface
+
+	apiBindingIndexer cache.Indexer
+
+	placementLister schedulinglisters.PlacementLister
+}
+
+// enqueue adds the logical cluster to the queue.
+func (c *controller) enqueue(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	name, clusterAwareName, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	clusterName, _ := clusters.SplitClusterAwareKey(clusterAwareName)
+
+	klog.Infof("Enqueueing logical cluster %q because of %T %s", clusterName, obj, name)
+	c.queue.Add(clusterName.String())
+}
+
+// Start starts the controller, which stops when ctx.Done() is closed.
+func (c *controller) Start(ctx context.Context, numThreads int) {
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting %s controller", controllerName)
+	defer klog.Infof("Shutting down %s controller", controllerName)
+
+	for i := 0; i < numThreads; i++ {
+		go wait.UntilWithContext(ctx, c.startWorker, time.Second)
+	}
+
+	<-ctx.Done()
+}
+
+func (c *controller) startWorker(ctx context.Context) {
+	for c.processNextWorkItem(ctx) {
+	}
+}
+
+func (c *controller) processNextWorkItem(ctx context.Context) bool {
+	// Wait until there is a new item in the working queue
+	k, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	key := k.(string)
+
+	// No matter what, tell the queue we're done with this key, to unblock
+	// other workers.
+	defer c.queue.Done(key)
+
+	if err := c.process(ctx, key); err != nil {
+		runtime.HandleError(fmt.Errorf("%q controller failed to sync %q, err: %w", controllerName, key, err))
+		c.queue.AddRateLimited(key)
+		return true
+	}
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *controller) process(ctx context.Context, key string) error {
+	clusterName := logicalcluster.New(key)
+
+	// check that binding exists, and create it if not
+	bindings, err := c.apiBindingIndexer.ByIndex(byWorkspace, clusterName.String())
+	if err != nil {
+		klog.Errorf("Failed to list bindings for workspace %q: %v", clusterName.String(), err)
+		return err
+	}
+
+	var locationWorkspace string
+	var apiBindingFound bool
+
+	for _, obj := range bindings {
+		binding := obj.(*apisv1alpha1.APIBinding)
+		if binding.Spec.Reference.Workspace == nil {
+			continue
+		}
+		if binding.Spec.Reference.Workspace.ExportName != reconcilerapiexport.TemporaryComputeServiceExportName {
+			continue
+		}
+
+		apiBindingFound = true
+		locationWorkspace = binding.Spec.Reference.Workspace.Path
+		break
+	}
+
+	if !apiBindingFound {
+		// do nothing if there is not apibinding for workload.
+		return nil
+	}
+
+	_, err = c.placementLister.Get(clusters.ToClusterAwareKey(clusterName, DefaultPlacementName))
```

Comment:
does this mean we enforce the creation of a "default" placement? I.e. we cannot even delete it later?

Wondering what a good logic is here. Only create it if there is no other placement existing? Or only create it on initial bind and if there is none? The later would be nice. Wondering how we can detect that we just bound.

Some cases:

1. we just bound, no placement exists => create default placement
2. we are bound, no placement exists because it had been deleted => don't do anything

Idea: we could annotate the APIBinding when we create the default placement. And if the annotation exists later, don't recreate it if it does not exist.